Imagine/Filter/Loader/DownscaleFilterLoader.php

```diff
@@ -30,7 +30,12 @@ public function load(ImageInterface $image, array $options = array())
             $widthRatio = $width / $origWidth;
             $heightRatio = $height / $origHeight;
 
-            $ratio = min($widthRatio, $heightRatio);
+            // faster check than is_null
+            if ($width === null || $height === null) {
+                $ratio = max($widthRatio, $heightRatio);
+            } else {
+                $ratio = min($widthRatio, $heightRatio);
+            }
```

Comment:
Of course, either way that is just a style thing, I don't think it will be a performance issue, so I'm okay with anything.